torchtune/modules/attention.py

```diff
@@ -282,7 +281,7 @@ def forward(
 
             # Update key-value cache
             if self.kv_cache is not None:
```

Comment:
What would be a sensible way to only use the cache for generation? For PPO, I need to run two forward passes with the same model, all under `no_grad`
1) During generation, with cache enabled
2) During trajectory estimation, using the same model with LoRA adapters disabled, but **without KV cacheing**

Then, we obviously don't want to use KV cacheing when `requires_grad=True` during the loss calculations.
My current solution is to wrap all generation code into `torch.inference_mode()`. Then, we check here for `if x.is_inference()` to only use the cache during inference, and not in any other case. 

Currently this is broken for compile (see issue here https://github.com/pytorch/pytorch/issues/135439#issuecomment-2337454492), but hopefully a fix by 2.5. Like most other places in the repo, could we version check for compile + generation?

@pbontrager @joecummings

Reply:
On reflection this isn't important for this PR and we can address in a follow up when it is needed for PPO.